### DISCUSSION TITLE
GTest with QNX RTOS

### DISCUSSION THREAD

**Discussion by salih-aydogan** [google/googletest#4669] [Community Help]
GTest is cross-compilable, which is especially useful for platforms like QNX. Also, the fact that there is a port support for QNX confirms that unit tests can be run on virtual machines with QNX installed. However, the situation regarding GMock support for integration tests seems a bit unclear. Can I use Gmock if I have an RTOS system when I want to test the integration between multiple modules?

**Comment by bmaxdk** [comment]
I am sure you can use GMock on a QNX RTOS to perform integration tests between multiple modules. Both GTest and GMock are designed to be portable and can be cross-compiled for various platforms, including RTOS environments like QNX. However, It will be tedious to make it work. You need to make minor modifications to the GMock source code or its build config to accommodate QNX-specific nuances.  Their usage of threading features that need to be supported by QNX and verifying that the QNX threading model is compatible with GMock's requirements and also c++ standard library implementation available on QNX supports all the necessary features used by GMock will be challenging tho

**Reply by salih-aydogan** [reply]
Thanks for your feedback. Is there another framework you could suggest to help you overcome this situation? How can I run integration tests in the QNX RTOS environment?

**Reply by bmaxdk** [reply]
I can suggent [cpputest](https://cpputest.github.io/). Since ti is easily cross-compilable for QNX and also supports both unit and integration tests with mocking capabilities, I think this can be good to try. I am sure there must be better frameworks for sure.
